tests_e2e/tests/scripts/agent_cgroups-check_cgroups_agent.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env pypy3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import logging
+import os
+import re
+import sys
+
+from tests_e2e.tests.lib.agent_log import AgentLog
+from tests_e2e.tests.lib.cgroup_helpers import BASE_CGROUP, AGENT_CONTROLLERS, get_agent_cgroup_mount_path, \
+    AGENT_SERVICE_NAME, exit_if_cgroups_not_supported, print_processes, print_cgroups, \
+    verify_agent_cgroup_assigned_correctly
+
+
+def verify_cgroup_controllers_on_disk():
+    logging.info("===== Verifying cgroup controllers exist on disk =====")
+
+    all_controllers_present = os.path.exists(BASE_CGROUP)
+
+    for controller in AGENT_CONTROLLERS:
+        controller_path = os.path.join(BASE_CGROUP, controller)
+        if not os.path.exists(controller_path):
+            logging.info('\tcould not verify controller %s', controller_path)
+            all_controllers_present = False
+        else:
+            logging.info('\tverified controller %s', controller_path)
+
+    if not all_controllers_present:
+        raise Exception('Unexpected cgroup controller status!')
```

Comment:
useless error message... we should include at least the controllers that we expect to find and were not found